src/literal.h

```diff
@@ -131,10 +193,37 @@ class Literal {
         WASM_UNREACHABLE("unexpected type");
     }
   }
-
-  static Literals makeZero(Type type);
-  static Literal makeSingleZero(Type type);
-
+  static Literal makeSignedMin(Type type) {
+    switch (type.getBasic()) {
+      case Type::i32:
+        return Literal(std::numeric_limits<int32_t>::min());
+      case Type::i64:
+        return Literal(std::numeric_limits<int64_t>::min());
+      default:
+        WASM_UNREACHABLE("unexpected type");
+    }
+  }
+  static Literal makeSignedMax(Type type) {
+    switch (type.getBasic()) {
+      case Type::i32:
+        return Literal(std::numeric_limits<int32_t>::max());
+      case Type::i64:
+        return Literal(std::numeric_limits<int64_t>::max());
+      default:
+        WASM_UNREACHABLE("unexpected type");
+    }
+  }
+  static Literal makeUnsignedMin(Type type) { return makeSingleZero(type); }
```

Comment:
Having a `makeUnsignedMin` seems to me like more trouble than it's worth if it also requires a separate `makeSingleZero` and `isSingleZero`. Would you be ok with just having `makeSignedMax`,` makeSignedMin`, `makeUnsignedMax`, and the normal `makeZero` plus the `is...` versions?

Reply:
Sure